packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,908 @@
+        triggerElement.removeEventListener(
+            'pointerleave',
+            this.handlePointerleave
+        );
+        this.removeEventListener(
+            'pointerleave',
+            this.handleOverlayPointerleave
+        );
+        triggerElement.addEventListener('pointerdown', this.handlePointerdown);
+        triggerElement.removeEventListener('keydown', this.handleKeydown);
+        triggerElement.removeEventListener('keyup', this.handleKeyup);
+        triggerElement.removeEventListener('longpress', this.handleLongpress);
+    }
+
+    protected bindEvents(
+        triggerElement: HTMLElement | VirtualTrigger | null
+    ): void {
+        if (!this.hasNonVirtualTrigger) return;
+        const nextTriggerElement = triggerElement as HTMLElement;
+        switch (this.triggerInteraction) {
+            case 'click':
+                this.bindClickEvents(nextTriggerElement);
+                return;
+            case 'longpress':
+                this.bindLongpressEvents(nextTriggerElement);
+                return;
+            case 'hover':
+                this.bindHoverEvents(nextTriggerElement);
+                return;
+        }
+    }
+
+    protected bindClickEvents(triggerElement: HTMLElement): void {
+        triggerElement.addEventListener('click', this.handleClick);
+        triggerElement.addEventListener(
+            'pointerdown',
+            this.handlePointerdownForClick
+        );
+    }
+
+    protected bindLongpressEvents(triggerElement: HTMLElement): void {
+        triggerElement.addEventListener('pointerdown', this.handlePointerdown);
+        triggerElement.addEventListener('keydown', this.handleKeydown);
+        triggerElement.addEventListener('keyup', this.handleKeyup);
+        triggerElement.addEventListener('longpress', this.handleLongpress);
+
+        this.prepareLongpressDescription(triggerElement);
+    }
+
+    protected bindHoverEvents(triggerElement: HTMLElement): void {
+        triggerElement.addEventListener('focusin', this.handleFocusin);
+        triggerElement.addEventListener('focusout', this.handleFocusout);
+        triggerElement.addEventListener(
+            'pointerenter',
+            this.handlePointerenter
+        );
+        triggerElement.addEventListener(
+            'pointerleave',
+            this.handlePointerleave
+        );
+        this.addEventListener('pointerleave', this.handleOverlayPointerleave);
+        if (this.receivesFocus === 'true') return;
+
+        this.prepareAriaDescribedby(triggerElement);
+    }
+
+    protected manageTriggerElement(triggerElement: HTMLElement | null): void {
+        if (triggerElement) {
+            this.unbindEvents(triggerElement);
+            this.releaseAriaDescribedby();
+        }
+        if (
+            !this.triggerElement ||
+            !!(this.triggerElement as VirtualTrigger).updateBoundingClientRect
+        ) {
+            return;
+        }
+        this.bindEvents(this.triggerElement);
+    }
```

Comment:
Do these annotations capture the goal of the function accurately?

```suggestion
    protected manageTriggerElement(triggerElement: HTMLElement | null): void {
        if (triggerElement) {
            this.unbindEvents(triggerElement);
            this.releaseAriaDescribedby();
        }
        const missingOrVirtual = !this.triggerElement || this.triggerElement instanceof VirtualTrigger;
        if (missingOrVirtual) {
            return;
        }
        this.bindEvents(this.triggerElement);
    }
```